### PR TITLE
update ETCDUnitSize IN KubeInKubeConfig for virtualcluster design

### PR DESCRIPTION
**Pull request by gao12312**
#### What type of PR is this?
/kind feature

#### What does this PR do?
update ETCDUnitSize IN KubeInKubeConfig for virtualcluster design

#### Which issue(s) does this PR fix?

Fixes #
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: deploy/crds/kosmos.io_virtualclusters.yaml [hunk 1]
```diff
@@ -43,6 +43,9 @@ spec:
           spec:
             description: Spec is the specification for the behaviour of the VirtualCluster.
             properties:
+              etcdSize:
+                description: definites the EtcdSize
+                type: string
               externalIP:
```
Comment (OrangeBao) [inline deploy/crds/kosmos.io_virtualclusters.yaml]: This file needs to be regenerated

File: pkg/kubenest/controlplane/etcd.go [hunk 2]
```diff
@@ -36,13 +36,30 @@ func DeleteVirtualClusterEtcd(client clientset.Interface, name, namespace string
...
-	resourceQuantity, err := resource.ParseQuantity(kubeNestConfiguration.KubeInKubeConfig.ETCDUnitSize)
-	if err != nil {
-		klog.Errorf("Failed to parse quantity %s: %v", kubeNestConfiguration.KubeInKubeConfig.ETCDUnitSize, err)
-		return err
+	var resourceQuantity resource.Quantity
+	var err error
+
+	if vc.Spec.KubeInKubeConfig.ETCDUnitSize != "" {
+		resourceQuantity, err = resource.ParseQuantity(vc.Spec.KubeInKubeConfig.ETCDUnitSize)
+		if err != nil {
```
Comment (OrangeBao) [inline pkg/kubenest/controlplane/etcd.go]: vc.Spec.KubeInKubeConfig may be nil

**Comment by duanmengkk** [comment]
rebase the commit first

**Pull request closed without merging**